File: C++/985. Sum of Even Numbers After Queries.cpp
/*
https://leetcode.com/problems/sum-of-even-numbers-after-queries/

We have an array A of integers, and an array queries of queries.

For the i-th query val = queries[i][0], index = queries[i][1], we add val to A[index].  Then, the answer to the i-th query is the sum of the even values of A.

(Here, the given index = queries[i][1] is a 0-based index, and each query permanently modifies the array A.)

Return the answer to all queries.  Your answer array should have answer[i] as the answer to the i-th query.

*/

class Solution {
public:    
    vector<int> sumEvenAfterQueries(vector<int>& A, vector<vector<int>>& queries) {
       int n = A.size(), sum = 0, idx = 0;; 
       vector<int> ret(n, 0);

        for(auto x : A)
            if(!(x&1))
                sum += x;

        for(auto q : queries){

            if(!(A[q[1]]&1))
                sum -= A[q[1]];        

            A[q[1]] += q[0];

            if(!(A[q[1]]&1))
                sum += A[q[1]];        

            ret[idx++] = sum;
        }    

        return ret;        
    }
};

File: Python/985. Sum of Even Numbers After Queries.py
''' 
985. Sum of Even Numbers After Queries
https://leetcode.com/problems/sum-of-even-numbers-after-queries/

You are given an integer array nums and an array queries where queries[i] = [vali, indexi].

For each query i, first, apply nums[indexi] = nums[indexi] + vali, then print the sum of the even values of nums.

Return an integer array answer where answer[i] is the answer to the ith query.
'''
class Solution:
    def sumEvenAfterQueries(self, nums, queries):
        arr_sum = 0
        res = []
        # Calculate the initial sum of the even values in the array
        for num in nums:
            if num % 2 == 0:
                arr_sum += num
        
        for query in queries:
            # If the value on which query is being performed is even remove value from sum of the array
            if nums[query[1]] % 2 == 0:
                arr_sum -= nums[query[1]]
            # Perform the specified query
            nums[query[1]] += query[0]
            # If the value after the query is even add the value back to sum of array
            if nums[query[1]] % 2 == 0:
                arr_sum += nums[query[1]]
            # Append the resulting arr_sum after the query has been performed
            res.append(arr_sum)
        return res
            